From f3ab017a229c9eca400a55d24cd9a4f759c4baa8 Mon Sep 17 00:00:00 2001
From: Stephen Finucane <stephenfin@redhat.com>
Date: Sat, 29 Aug 2026 11:27:47 +0100
Subject: [PATCH] json-rpc: Rename TLS options

These options control TLS configuration for the JSON RPC server.
keystoneauth 5.17.0 adds new, identically-named options that control TLS
configuration for the *client*. Allow users to distinguish between the
two by adding a `server_` prefix to the former. Aliases are provided,
though these introduce a small upgrade impact: users who had overridden
these options to control server configuration will see the same
configuration inherited for clients (or rather, will as soon as ksa is
bumped). This will be covered via an update to the release note in a
subsequent change.

Runtime-only backport: upstream unit tests and release note are omitted.

Change-Id: I1972c85de47fc70758753665488f822b4320a264
Signed-off-by: Stephen Finucane <stephenfin@redhat.com>
---
 ironic/common/wsgi_service.py |    4 ++--
 ironic/conf/api.py            |    6 ++++--
 ironic/conf/json_rpc.py       |    6 ++++--
 3 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/ironic/common/wsgi_service.py b/ironic/common/wsgi_service.py
index b624916325..1e004fb92c 100644
--- a/ironic/common/wsgi_service.py
+++ b/ironic/common/wsgi_service.py
@@ -92,14 +92,14 @@ def __init__(self, name, app, conf, use_ssl=None):

             validate_cert_paths(cert_file, key_file)

-            tls_ciphers = conf.tls_ciphers
+            tls_ciphers = conf.server_tls_ciphers
             self.server.ssl_adapter = cheroot_ssl.BuiltinSSLAdapter(
                 certificate=cert_file,
                 private_key=key_file,
                 ciphers=tls_ciphers,
             )

-            tls_min = conf.tls_minimum_version
+            tls_min = conf.server_tls_minimum_version
             if tls_min:
                 tls_utils.check_tls_version_supported(tls_min)
                 version = tls_utils.TLS_VERSION_MAP[tls_min]
diff --git a/ironic/conf/api.py b/ironic/conf/api.py
index 0b9fe5c9c5..567c74138c 100644
--- a/ironic/conf/api.py
+++ b/ironic/conf/api.py
@@ -121,7 +121,8 @@ def __call__(self, value):
                help="Private key file to use when starting "
                     "the server securely."),
     cfg.StrOpt(
-        'tls_minimum_version',
+        'server_tls_minimum_version',
+        deprecated_name='tls_minimum_version',
         default='1.3',
         choices=[
             ('1.2',
@@ -141,7 +142,8 @@ def __call__(self, value):
                "external service such as a reverse proxy "
                "or a WSGI application container.")),
     cfg.StrOpt(
-        'tls_ciphers',
+        'server_tls_ciphers',
+        deprecated_name='tls_ciphers',
         help=_("Set the list of available ciphers for the "
                "integrated API server. The value should be "
                "a string in the OpenSSL cipher list format. "
diff --git a/ironic/conf/json_rpc.py b/ironic/conf/json_rpc.py
index 10c97556e3..5a44da3f6c 100644
--- a/ironic/conf/json_rpc.py
+++ b/ironic/conf/json_rpc.py
@@ -49,7 +49,8 @@
     cfg.StrOpt('key_file',
                help=_("Private key file matching cert_file.")),
     cfg.StrOpt(
-        'tls_minimum_version',
+        'server_tls_minimum_version',
+        deprecated_name='tls_minimum_version',
         default='1.3',
         choices=[
             ('1.2',
@@ -66,7 +67,8 @@
                "TLS 1.3 since TLS 1.0 and 1.1 are "
                "deprecated (RFC 8996).")),
     cfg.StrOpt(
-        'tls_ciphers',
+        'server_tls_ciphers',
+        deprecated_name='tls_ciphers',
         help=_("Set the list of available ciphers for the "
                "JSON-RPC server. The value should be a "
                "string in the OpenSSL cipher list format. "
